data-prepper-plugins/opensearch/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java

```diff
@@ -1268,6 +1268,42 @@ public void testOpenSearchDynamicIndexWithDate(final String testIndex, final Str
         sink.shutdown();
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "id, yyyy-MM-dd, %{yyyy-MM-dd}-test-${id}-index",
+            "id, yyyy-MM-dd, test-%{yyyy-MM-dd}-${id}-index",
+            "id, yyyy-MM-dd, test-${id}-%{yyyy-MM-dd}-index",
+    })
+    public void testOpenSearchDynamicIndexWithDateNotAsSuffix(
+            final String testIndex, final String testDatePattern, final String dynamicTestIndexAlias) throws IOException {
+        final String testIndexName = "idx1";
+        SimpleDateFormat formatter = new SimpleDateFormat(testDatePattern);
+        Date date = new Date();
+        String expectedDate = formatter.format(date);
+        final String expectedIndexAlias = dynamicTestIndexAlias
+                .replace("%{yyyy-MM-dd}", expectedDate)
+                .replace("${id}", testIndexName);
+        final String data = UUID.randomUUID().toString();
+        final Map<String, Object> dataMap = Map.of("data", data);
+        final Event testEvent = JacksonEvent.builder()
+                .withData(dataMap)
+                .withEventType("event")
+                .build();
+        testEvent.put(testIndex, testIndexName);
+
+        Map<String, Object> expectedMap = testEvent.toMap();
+
+        final List<Record<Event>> testRecords = Collections.singletonList(new Record<>(testEvent));
+
+        final PluginSetting pluginSetting = generatePluginSetting(null, dynamicTestIndexAlias, null);
+        final OpenSearchSink sink = createObjectUnderTest(pluginSetting, true);
+        sink.output(testRecords);
+        final List<Map<String, Object>> retSources = getSearchResponseDocSources(expectedIndexAlias);
```

Comment:
Thanks for these tests. We should also have a test case to verify that index templates are created correctly and applied correctly.

Reply:
Thanks for the suggestion. I missed the part about index patterns. Corrected it and added integ tests in most recent commits.